### PR TITLE
mcs: support ttl config

### LINKED ISSUE #7296: mcs: support to sync config with ttl
## Enhancement Task

cannot run test in api mode 
https://github.com/tikv/pd/pull/7278#discussion_r1378802048

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Close #7296

### What is changed and how does it work?

This PR supports TTL config for scheduling service.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/mcs/scheduling/server/server.go [hunk 1]
```diff
@@ -534,7 +535,7 @@ func CreateServer(ctx context.Context, cfg *config.Config) *Server {
 		BaseServer:        server.NewBaseServer(ctx),
 		DiagnosticsServer: sysutil.NewDiagnosticsServer(cfg.Log.File.Filename),
 		cfg:               cfg,
-		persistConfig:     config.NewPersistConfig(cfg),
+		persistConfig:     config.NewPersistConfig(cfg, cache.NewStringTTL(ctx, time.Second*5, time.Minute*5)),
```
Comment (lhy1024) [inline pkg/mcs/scheduling/server/server.go:538]: how about using a const value

File: server/config/persist_options.go [hunk 2]
```diff
@@ -852,16 +835,22 @@ func (o *PersistOptions) GetMinResolvedTSPersistenceInterval() time.Duration {
...
 // SetTTLData set temporary configuration
 func (o *PersistOptions) SetTTLData(parCtx context.Context, client *clientv3.Client, key string, value string, ttl time.Duration) error {
 	if o.ttl == nil {
 		o.ttl = cache.NewStringTTL(parCtx, time.Second*5, time.Minute*5)
 	}
-	_, err := etcdutil.EtcdKVPutWithTTL(parCtx, client, ttlConfigPrefix+"/"+key, value, int64(ttl.Seconds()))
-	if err != nil {
-		return err
+	if ttl != 0 {
+		// the minimum ttl is 5 seconds, if the given ttl is less than 5 seconds, we will use 5 seconds instead.
```
Comment (lhy1024) [inline server/config/persist_options.go:844]: why add minium?

Comment (rleungx) [reply server/config/persist_options.go:844]: It's set by etcd

File: pkg/mcs/scheduling/server/server.go (hunk 1 above)
Comment (lhy1024) [reply pkg/mcs/scheduling/server/server.go:538]: Will we do it?

**Review by lhy1024 — COMMENTED** [review]
the rest LGTM

File: pkg/mcs/scheduling/server/server.go (hunk 1 above)
Comment (rleungx) [reply pkg/mcs/scheduling/server/server.go]: done

**Review by lhy1024 — APPROVED** [review]

File: tests/server/config/config_test.go [hunk 3]
```diff
@@ -491,11 +491,11 @@ func (suite *configTestSuite) assertTTLConfigItemEqaul(
 		}
 		return false
 	}
-	suite.True(checkfunc(cluster.GetLeaderServer().GetServer().GetPersistOptions()))
+	suite.True(checkFunc(cluster.GetLeaderServer().GetServer().GetPersistOptions()))
 	if cluster.GetSchedulingPrimaryServer() != nil {
 		// wait for the scheduling primary server to be synced
```
Comment (CabinfeverB) [inline tests/server/config/config_test.go:496]: The check here does not include ttl expired cases? Because there was no waiting.

**Review by CabinfeverB — COMMENTED** [review]
rest LGTM

**Pull request merged**